packages/core/android/src/main/kotlin/src/DdSdkReactNativePackage.kt

```diff
@@ -0,0 +1,78 @@
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.reactnative
+
+import com.facebook.react.TurboReactPackage
+import com.facebook.react.bridge.NativeModule
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.module.model.ReactModuleInfo
+import com.facebook.react.module.model.ReactModuleInfoProvider
+
+/**
+ * Package of native dd-sdk-reactnative native modules.
+ */
+class DdSdkReactNativePackage : TurboReactPackage() {
+    override fun getModule(name: String, reactContext: ReactApplicationContext): NativeModule? {
+        return if (name == DdSdkImplementation.NAME) {
+            DdSdk(reactContext)
+        } else if (name == DdRumImplementation.NAME) {
+            DdRum(reactContext)
+        } else if (name == DdTraceImplementation.NAME) {
+            DdTrace(reactContext)
+        } else if (name == DdLogsImplementation.NAME) {
+            DdLogs(reactContext)
+        } else {
+            null
+        }
+    }
+
+    override fun getReactModuleInfoProvider(): ReactModuleInfoProvider {
+        return ReactModuleInfoProvider {
+            val moduleInfos: MutableMap<String, ReactModuleInfo> =
+                HashMap()
+            val isTurboModule: Boolean = BuildConfig.IS_NEW_ARCHITECTURE_ENABLED
+            moduleInfos[DdSdkImplementation.NAME] = ReactModuleInfo(
+                DdSdkImplementation.NAME,
+                DdSdkImplementation.NAME,
+                false,  // canOverrideExistingModule
+                false,  // needsEagerInit
+                true,  // hasConstants
+                false,  // isCxxModule
+                isTurboModule // isTurboModule
+            )
+            moduleInfos[DdRumImplementation.NAME] = ReactModuleInfo(
+                DdRumImplementation.NAME,
+                DdRumImplementation.NAME,
+                false,  // canOverrideExistingModule
+                false,  // needsEagerInit
+                true,  // hasConstants
+                false,  // isCxxModule
+                isTurboModule // isTurboModule
+            )
+            moduleInfos[DdTraceImplementation.NAME] = ReactModuleInfo(
+                DdTraceImplementation.NAME,
+                DdTraceImplementation.NAME,
+                false,  // canOverrideExistingModule
+                false,  // needsEagerInit
+                true,  // hasConstants
+                false,  // isCxxModule
+                isTurboModule // isTurboModule
+            )
+            moduleInfos[DdLogsImplementation.NAME] = ReactModuleInfo(
+                DdLogsImplementation.NAME,
+                DdLogsImplementation.NAME,
+                false,  // canOverrideExistingModule
+                false,  // needsEagerInit
+                true,  // hasConstants
+                false,  // isCxxModule
+                isTurboModule // isTurboModule
+            )
```

Comment:
I would rather simplify it to the following given that params for `ReactModuleInfo` are always the same (except the name). And maybe mutable collection above is not needed at all.

```suggestion
            val moduleInfos = listOf(
              DdSdkImplementation.NAME,
              DdRumImplementation.NAME,
              DdTraceImplementation.NAME,
              DdLogsImplementation.NAME
            ).associateWith {
              ReactModuleInfo(
                it,
                it,
                false,  // canOverrideExistingModule
                false,  // needsEagerInit
                true,  // hasConstants
                false,  // isCxxModule
                isTurboModule // isTurboModule
              )
            }
```